app/routers/api/remote.py

```
@@ -1407,8 +1406,11 @@ async def serve_transfer_request(
     except ValueError as e:
         raise HTTPException(status_code=400, detail=str(e)) from e
 
-    # Queue the transfer to run in background
-    background_tasks.add_task(remote_transfer_service.run_transfer, job.id)
+    # Detach the transfer from the request lifecycle so shutdown is not blocked
+    # waiting for Starlette background tasks to finish a long file upload.
+    import asyncio
+
+    asyncio.create_task(remote_transfer_service.run_transfer(job.id))
```

Ruff (0.15.11)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)